coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -269,45 +305,149 @@ class SingleClusterPlanner(val dataset: Dataset,
     */
   // scalastyle:off cyclomatic.complexity
   override def walkLogicalPlanTree(logicalPlan: LogicalPlan,
-                                  qContext: QueryContext): PlanResult = {
+                                  qContext: QueryContext,
+                                   forceInProcess: Boolean): PlanResult = {
```

Comment:
> that it will cause the exec plan subtree for the entire logical plan to be pushed down fully into individual shards and concatenated

Not true-- if some`ExecPlan` tree pulls data from multiple shards, materialization with `forceInProcess=true` does not affect whether-or-not a pushdown occurs, and it will blindly dispatch everything with an `InProcessDispatcher`. 

I had added a small blurb about it with the other param docs in `DefaultPlanner::defaultWalkLogicalPlanTree`, but I've now moved all of these descriptions to (hopefully more appropriately) the abstract `DefaultPlanner::walkLogicalPlanTree`.

(The `forceInProcess` description is:
```
@param forceInProcess if true, all materialized plans will dispatch via an InProcessDispatcher
```
Let me know if that is insufficient.)

Reply:
sounds good. Add 'for this logical plan'

```
@param forceInProcess if true, all materialized plans for this entire logical plan will dispatch via an InProcessDispatcher
```

Reply:
added